File: leetcode-top-interview-150/121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        pf = 0
        minP = prices[0]

        for price in prices:
            if price < minP:
                minP = price
            else:
                pf = max(pf, price - minP)

        return pf

File: leetcode-top-100-liked/121.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
    
        var maxVal = 0
        var buyVal = prices[0]

        for(i in 1..prices.lastIndex) {
            var curVal = prices[i]

            if(curVal < buyVal) {
                buyVal = curVal
            } else {
                maxVal = max(maxVal, curVal - buyVal)
            }
        }
        return maxVal
    }
}